src/subapps/admin/components/Settings/DangerZoneSubView.tsx

```diff
@@ -233,11 +303,50 @@ const DangerZoneSubView = ({ project }: Props) => {
             htmlType="button"
             onClick={handleOpenDeleteModal}
           >
+            <DeleteOutlined />
             Delete this Project
           </Button>
         </AccessControl>
       ),
     },
+    {
+      key: 'undo-deprecate-project-section',
+      title: 'Undo Deprecation of this Project',
+      description: 'Restore this project to its active state.',
+      action: (
+        <AccessControl
+          path={[`${orgLabel}/${projectLabel}`]}
+          permissions={['projects/write']}
+          noAccessComponent={() => (
+            <Tooltip title="You have no permissions to undo the deprecation of this project">
+              <Button disabled style={{ margin: 0, marginRight: 10 }}>
+                <span>Undo Deprecation</span>
+                <HasNoPermission />
+              </Button>
+            </Tooltip>
+          )}
+        >
+          <Button
+            style={{ margin: 0, marginRight: 10 }}
+            type="primary"
+            htmlType="button"
+            disabled={!project._deprecated} // Enable button only if project is deprecated
+            onClick={() =>
```

Comment:
the onClick can be pure fn since it's has unique use case and one place, meaning, it can take only fn without input data since the data is reside inside the component